google_guest_agent/oslogin.go

```diff
@@ -265,6 +265,14 @@ func filterGoogleLines(contents string) []string {
 			filtered = append(filtered, line)
 		}
 	}
+	// Unix text files should end with a final "\n"
+	// which strings.Split will account for with a terminal ""
+	// so we remove that here, to avoid adding more empty lines.
+	// But we can't assume that every file does end in a newline,
+	// so only remove it if it's empty as expected.
+	if len(filtered) > 0 && filtered[len(filtered)-1] == "" {
```

Comment:
Did you mean to compare with something other than the empty string here? I don't think `filtered[len(filtered)-1] == ""` will ever be true.

Reply:
A split on a string with empty lines will yield empty string elements. The `\n` is removed when splitting.

So yes, this is intentional. It catches the case where the file ends in a newline, because split includes the empty string _after_ that terminal newline.

Reply:
To clarify a bit more, that block is there to avoid accidentally deleting the last element when there is _not_ a terminal newline (in which case the final element might not be empty).

Reply:
Ah, nice, I'd missed that `filtered` was an array of strings, not simply a string. Sounds good.